### PR TITLE
enhance: add the related data size for the read apis

### PR DESCRIPTION
**Pull request by SimFG**
issue: #30436
origin pr: #30438
related pr: #31772

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -88,6 +88,7 @@ const (
 	segmentStateLabelName    = "segment_state"
 	segmentIDLabelName       = "segment_id"
 	segmentLevelLabelName    = "segment_level"
+	databaseName             = "database_name"
```
Comment (jaime0815) [inline pkg/metrics/metrics.go:91]: there has already been defined a database name label.

File: pkg/metrics/proxy_metrics.go [hunk 2]
```diff
@@ -263,6 +263,15 @@ var (
 			Help:      "count of bytes sent back to sdk",
 		}, []string{nodeIDLabelName})
 
+	// ProxyReportValue records value about the request
+	ProxyReportValue = prometheus.NewCounterVec(
```
Comment (jaime0815) [inline pkg/metrics/proxy_metrics.go:267]: We'd better clean up metrics after dropping the database.

File: internal/querynodev2/tasks/query_task.go [hunk 3]
```diff
@@ -125,6 +126,10 @@ func (t *QueryTask) Execute() error {
 		return err
 	}
 
+	relatedDataSize := lo.Reduce(querySegments, func(acc int64, seg segments.Segment, _ int) int64 {
+		return acc + seg.MemSize()
```
Comment (jaime0815) [inline internal/querynodev2/tasks/query_task.go:130]: note that the actual data size will be larger than on disk size if we use the memory size

File: internal/proxy/impl.go [hunk 4]
```diff
@@ -2685,7 +2706,8 @@ func (node *Proxy) Search(ctx context.Context, request *milvuspb.SearchRequest)
 		Status: merr.Success(),
 	}
 	err2 := retry.Handle(ctx, func() (bool, error) {
-		rsp, err = node.search(ctx, request)
+		rsp, err = node.
```
Comment (czs007) [inline internal/proxy/impl.go:2709]: Why is there a line break here? Typo?

**Pull request merged**